modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/GetDataStreamsTransportAction.java

```diff
@@ -121,7 +120,7 @@ protected void masterOperation(
         List<String> results = DataStreamsActionUtil.getDataStreamNames(iner, clusterState, request.getNames(), request.indicesOptions());
         Map<String, DataStream> dataStreams = clusterState.metadata().dataStreams();
 
-        return results.stream().map(dataStreams::get).sorted(Comparator.comparing(DataStream::getName)).collect(Collectors.toList());
+        return results.stream().map(dataStreams::get).sorted(Comparator.comparing(DataStream::getName)).toList();
```

Comment:
Previous `.collect(toList())` is mutable while `.toList()` is not. Is it safe to replace it everywhere?

Reply:
It wasn't in the one spot in `SnapshotsService` that I fixed but otherwise I couldn't find any problems. It's rare that calling code assumes mutability on this kind of API and it shouldn't IMO.